dashboard/src/components/AppList/AppList.test.tsx

```diff
@@ -25,14 +26,15 @@ describe("while fetching apps", () => {
     props = { ...defaultProps, apps: { isFetching: true } };
   });
 
-  it("matches the snapshot", () => {
-    const wrapper = shallow(<AppList {...props} />);
-    expect(wrapper).toMatchSnapshot();
+  afterAll(() => {
+    // Calling the shared specs here so they are run after all the beforeEach
+    // callbacks have been run which set the default props and prop
+    sharedSpecs.loadingSpecs(AppList, props);
```

Comment:
I find this a bit difficult to understand:

 - What does `specs` mean here? It seems that it is some kind of meta-test that can be reusable.
 - Why using `afterAll` instead of a `describe` or a simple `it`?
 - Shouldn't you set the property `isFetching: true` so you ensure that the Loading component should appear regardless of how `props` is modified outside the scope of this test?

Reply:
That's a good point, I do not find it ideal either.

spec == test, and what this does is just to run a set of shared tests. In a similar way rspec shared_examples work. It seems that jest does not support it natively so I based it on https://medium.com/@walreyes/sharing-specs-in-jest-82864d4d5f9e (without adding the behaves_like)

> Why using afterAll instead of a describe or a simple it?

I need to add it inside an afterAll to have access to the properties set in the other callbacks. w.r.t `it` the linter (I think) was complaining about nested its. This can be solved by moving the props setting outside of the callback and as part of the describe. The only reason is there is because it guarantees that the vble has not been mutated,

> Shouldn't you set the property isFetching: true so you ensure that the Loading component should appear regardless of how props is modified outside the scope of this test?

I want to know more, isFetching is set to true no? https://github.com/kubeapps/kubeapps/pull/637/files/7d4aa8e593537f47c46d090c7e23dee35b0508f0#diff-bad52a69fd8cd98a31a6337371ae22b9R26

I have another patch that changes this code a bit so I'll prefer to land this one and handle this there. I might end up removing this shared example after all.